lib/uklibparam/libparam.lds.S

```diff
@@ -4,9 +4,8 @@ SECTIONS {
 	. = ALIGN(8);
 	.UK_LIBPARAM_PARAMSECTION_NAME : {
 		PROVIDE(UK_LIBPARAM_PARAMSECTION_STARTSYM = .);
-		KEEP(*(.UK_LIBPARAM_PARAMSECTION_NAME))
-		KEEP(*(.UK_LIBPARAM_PARAMSECTION_NAME.*))
+		KEEP(*(SORT_BY_NAME(UK_LIBPARAM_PARAM_NAMEPREFIX*)));
```

Comment:
Strange, even though this has `KEEP` and I have environ, netdev and fstab enabled, which should have registered something in this section, it seems to be empty and non-existent:
```
$ readelf -S --wide build/elfloader_qemu-x86_64.dbg           
There are 33 section headers, starting at offset 0x55ccb8:                                
                                                                                          
Section Headers:                                                                          
  [Nr] Name              Type            Address          Off    Size   ES Flg Lk Inf Al  
  [ 0]                   NULL            0000000000000000 000000 000000 00      0   0  0  
  [ 1] .text             PROGBITS        0000000000100000 001000 0c65ae 00  AX  0   0 4096
  [ 2] .rodata           PROGBITS        00000000001c7000 0c8000 016440 00   A  0   0 32  
  [ 3] .uk_fs_list       PROGBITS        00000000001dd440 0de440 000010 00  WA  0   0  8  
  [ 4] .uk_thread_inittab PROGBITS        00000000001dd450 0de450 000048 00   A  0   0  8 
  [ 5] .uk_excpttab      PROGBITS        00000000001dd498 0de498 000090 00   A  0   0  1  
  [ 6] .uk_libinfo       PROGBITS        00000000001dd528 0de528 000de5 00   A  0   0  1  
  [ 7] .uk_posix_clonetab PROGBITS        00000000001de310 0df310 000140 00   A  0   0  8 
  [ 8] .eh_frame         PROGBITS        00000000001df000 0e0000 00a4d8 00   A  0   0  8  
  [ 9] .eh_frame_hdr     PROGBITS        00000000001e94d8 0ea4d8 0007b4 00   A  0   0  4  
  [10] .uk_ctortab       PROGBITS        00000000001ea000 0eb000 000060 00   A  0   0  8  
  [11] .uk_inittab       PROGBITS        00000000001ea060 0eb060 0000c0 00   A  0   0  8  
  [12] .uk_eventtab      PROGBITS        00000000001ea120 0eb120 0000a0 00  WA  0   0  8  
  [13] .init_array       INIT_ARRAY      00000000001ea1c0 0eb1c0 000008 08  WA  0   0  8  
  [14] .tdata            PROGBITS        00000000001eb000 0ec000 000004 00 WAT  0   0  8  
  [15] .tbss             NOBITS          00000000001eb008 0ec004 000028 00 WAT  0   0  8  
  [16] .data             PROGBITS        00000000001ec000 0ed000 001bc8 00  WA  0   0 32  
  [17] .posix_socket_driver_list PROGBITS        00000000001edbc8 0eebc8 000050 00  WA  0 
  0  8                                                                                    
  [18] .uk_bootinfo      PROGBITS        00000000001edc18 0eec18 000c50 00  WA  0   0  8  
  [19] .bss              NOBITS          00000000001ef000 0ef868 464000 00  WA  0   0 64  
  [20] .comment          PROGBITS        0000000000000000 0ef868 000ba0 01  MS  0   0  1  
  [21] .debug_aranges    PROGBITS        0000000000000000 0f0410 011390 00      0   0 16  
  [22] .debug_info       PROGBITS        0000000000000000 1017a0 13c48b 00      0   0  1  
  [23] .debug_abbrev     PROGBITS        0000000000000000 23dc2b 025148 00      0   0  1  
  [24] .debug_line       PROGBITS        0000000000000000 262d73 07794e 00      0   0  1  
  [25] .debug_frame      PROGBITS        0000000000000000 2da6c8 02a2b0 00      0   0  8  
  [26] .debug_str        PROGBITS        0000000000000000 304978 11fa9f 01  MS  0   0  1  
  [27] .debug_macro      PROGBITS        0000000000000000 424417 0fa3b0 00      0   0  1  
  [28] .debug_line_str   PROGBITS        0000000000000000 51e7c7 009fa6 01  MS  0   0  1  
  [29] .debug_rnglists   PROGBITS        0000000000000000 52876d 00b480 00      0   0  1  
  [30] .symtab           SYMTAB          0000000000000000 533bf0 019e90 18     31 3005  8 
  [31] .strtab           STRTAB          0000000000000000 54da80 00f0b5 00      0   0  1  
  [32] .shstrtab         STRTAB          0000000000000000 55cb35 00017f 00      0   0  1  
Key to Flags:                                                                             
  W (write), A (alloc), X (execute), M (merge), S (strings), I (info),                    
  L (link order), O (extra OS processing required), G (group), T (TLS),                   
  C (compressed), x (unknown), o (OS specific), E (exclude),                              
  D (mbind), l (large), p (processor specific)                                            
```
```
$ nm build/elfloader_qemu-x86_64.dbg | grep uk_libparam_param
00000000001edbc8 D uk_libparam_params_end                                                
00000000001edbc8 D uk_libparam_params_start                                              
```
Am I missing something?

Reply:
Did you enable `CONFIG_LIBUKNETDEV_EINFO_LIBPARAM`?

Reply:
Ok then this is very weird. Honestly, it feels like a static linker bug. I always thought these are equivalent.